sdk/storage/Azure.Storage.Common/src/Shared/LazyLoadingReadOnlyStream.cs

```diff
@@ -97,7 +108,13 @@ public async Task<int> ReadInternal(byte[] buffer, int offset, int count, bool a
 
             if (_position == _length)
             {
-                return 0;
+                // In case the blob grow since our last download call.
+                _length = await GetBlobLengthInternal(async, cancellationToken).ConfigureAwait(false);
```

Comment:
Nit - might be worth adding an extra `bool _allowBlobModifications = true;` flag that we change to false when forcing an `If-Match` to optionally spare this extra call when the blob isn't allowed to change from under us.

Reply:
Fixed in next iteration.